src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -151,12 +158,17 @@ public void configure(Map<String, ?> configs) {
         excludedPrincipalNameList = config.getExcludedPrincipalNameList();
 
         long storageCheckIntervalMillis = TimeUnit.SECONDS.toMillis(config.getStorageCheckInterval());
-        List<Path> logDirs = config.getLogDirs().stream().map(Paths::get).collect(Collectors.toList());
-        storageChecker.configure(storageCheckIntervalMillis,
-                logDirs,
-                this::updateUsedStorage);
-
-        log.info("Configured quota callback with {}. Storage quota (soft, hard): ({}, {}). Storage check interval: {}ms", quotaMap, storageQuotaSoft, storageQuotaHard, storageCheckIntervalMillis);
+        if (storageCheckerFuture != null) {
+            storageCheckerFuture.cancel(true);
+        }
+        if (storageCheckIntervalMillis > 0L) {
+            List<Path> logDirs = config.getLogDirs().stream().map(Paths::get).collect(Collectors.toList());
+            storageChecker.configure(
+                    logDirs,
+                    this::updateUsedStorage);
+            storageCheckerFuture = backgroundScheduler.scheduleAtFixedRate(storageChecker, storageCheckIntervalMillis, storageCheckIntervalMillis, TimeUnit.MILLISECONDS);
```

Comment:
I'm on the fence about using `atFixedRate` or `withFixedDelay` here. 

I think `fixedRate` matches the configuration intent but `withFixedDelay` would be a closer match to the previous implementation. Given we expect the task to execute very quickly the distinction is probably moot.

Reply:
I'd be inclined to fixed delay myself. That guarantees we can't end up checking continuously. But you're right that it's difficult to see that it matters in normal circumstances.

Reply:
I've switched to `withFixedDelay`.